packages/shared/src/components/notifications/Toast.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { ReactElement, useCallback, useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { useQuery, useQueryClient } from 'react-query';
+import {
+  ToastNotification,
+  TOAST_NOTIF_KEY,
+} from '../../hooks/useToastNotification';
+import classed from '../../lib/classed';
+import { Button } from '../buttons/Button';
+import styles from './Toast.module.css';
+import XIcon from '../../../icons/x.svg';
+import { isTouchDevice } from '../../lib/tooltip';
+
+interface ToastProps {
+  autoDismissNotifications?: boolean;
+}
+
+const Container = classed(
+  'div',
+  'fixed left-1/2 flex flex-col justify-center bg-theme-label-primary p-2 rounded-14 border-theme-divider-primary shadow-2',
+  styles.toastContainer,
+);
+const Content = classed('div', 'relative flex flex-row items-center ml-2');
+const Message = classed(
+  'div',
+  'flex-1 mr-2 typo-subhead text-theme-label-invert',
+);
+const Progress = classed(
+  'span',
+  styles.toastProgress,
+  'absolute -bottom-2 h-1 ease-in-out bg-theme-status-cabbage rounded-full',
+);
+const INTERVAL_COUNT = 10;
+const IN_OUT_ANIMATION = 140;
+const TEMPORARY_ID = 1;
+
+const Toast = ({
+  autoDismissNotifications = false,
+}: ToastProps): ReactElement => {
+  const router = useRouter();
+  const client = useQueryClient();
+  const [intervalId, setIntervalId] = useState<number>(null);
+  const [timer, setTimer] = useState(0);
+  const { data: toast } = useQuery<ToastNotification>(TOAST_NOTIF_KEY);
+  const setToast = (data: ToastNotification) =>
+    client.setQueryData(TOAST_NOTIF_KEY, data);
+
+  useEffect(() => {
+    if (toast && timer === 0 && intervalId) {
+      window.clearInterval(intervalId);
+      setIntervalId(null);
+      setTimeout(() => setToast(null), IN_OUT_ANIMATION);
```

Comment:
We probably also want to clear this timeout on cleanup.

Reply:
Indeed, this has to be cleaned up! 🔭